server/lib/matchmaking/matchmaking-service.ts

```diff
@@ -43,10 +44,10 @@ interface GameLoaderCallbacks {
     matchInfo: Match
     clients: List<ClientSocketsGroup>
     slots: List<Slot>
-    setup?: Partial<{
+    setup?: {
```

Comment:
Don't really understand this change, why move from `undefined` meaning "this thing is not present" to `random value I've chosen` serving that purpose?

Reply:
I thought `Partial` made all of the fields of the object optional, and seemed to me like `gameId` and `seed` are required to successfully start a game. Reverted it now though.

Reply:
It does. I don't get why this was ever optional in the first place and would be fine with you moving it away from being optional (and Partial) if you can verify that it was just a mistake. But moving from it defaulting to `{}` (and thus each of these values being `undefined`) to it defaulting to an object with dummy values for each of these keys (which is way harder and more confusing to verify as meaning to indicate "not present") was a bad change imo.